packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts

```diff
@@ -48,397 +28,167 @@ const supportedCurrencies: Record<string,  Record<RequestLogicTypes.CURRENCY, st
-  }
-
-  const copiedExtensionState: ExtensionTypes.IState = Utils.deepCopy(extensionState);
-
-  // update fee address and amount
-  copiedExtensionState.values.feeAddress = extensionAction.parameters.feeAddress;
-  copiedExtensionState.values.feeAmount = extensionAction.parameters.feeAmount;
-
-  // update events
-  copiedExtensionState.events.push({
-    name: ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_FEE,
-    parameters: {
-      feeAddress: extensionAction.parameters.feeAddress,
-      feeAmount: extensionAction.parameters.feeAmount,
-    },
-    timestamp,
-  });
-
-  return copiedExtensionState;
-}
-
-/**
- * Check if an ethereum address is valid
- *
- * @param {string} address address to check
- * @returns {boolean} true if address is valid
- */
-function isValidAddress(address: string): boolean {
-  return walletAddressValidator.validate(address, 'ethereum');
-}
+    };
+  }
+
+  /**
+   * Validates the payment network of the request currency.
+   *
+   * @param currency
+   */
+  protected validateSupportedCurrency(
+    request: RequestLogicTypes.IRequest,
+    extensionAction: ExtensionTypes.IAction,
+  ): void {
+    const network =
+      extensionAction.parameters.network ||
+      request.extensions[this.paymentNetworkId]?.values.network;
+
+    // Nothing can be validated if the network has not been given yet
+    if (!network) {
+      return;
+    }
 
-/**
- * Throw if a currency is not supported
- *
- * @param currency currency to check
- * @param network network of the payment
- */
-function checkSupportedCurrency(currency: RequestLogicTypes.ICurrency, network: string): void {
-  if (!supportedCurrencies[network]) {
-    throw new Error(`The network (${network}) is not supported for this payment network.`);
-  }
+    if (!supportedCurrencies[network]) {
+      throw new Error(`The network (${network}) is not supported for this payment network.`);
+    }
 
-  if (!supportedCurrencies[network][currency.type]) {
-    throw new Error(
-      `The currency type (${currency.type}) of the request is not supported for this payment network.`,
-    );
-  }
+    if (!supportedCurrencies[network][request.currency.type]) {
+      throw new Error(
+        `The currency type (${request.currency.type}) of the request is not supported for this payment network.`,
+      );
+    }
 
-  if (!supportedCurrencies[network][currency.type].includes(currency.value)) {
-    throw new Error(
-      `The currency (${currency.value}) of the request is not supported for this payment network.`,
-    );
+    if (!supportedCurrencies[network][request.currency.type].includes(request.currency.value)) {
+      throw new Error(
+        `The currency (${request.currency.value}) of the request is not supported for this payment network.`,
+      );
+    }
   }
 }
 
+const conversionErc20FeeProxyContract: ExtensionTypes.PnAnyToErc20.IAnyToERC20 = new anyToErc20ProxyPaymentNetwork() as ExtensionTypes.PnAnyToErc20.IAnyToERC20;
```

Comment:
this part is not required if the class implements the interface
```suggestion
const conversionErc20FeeProxyContract: ExtensionTypes.PnAnyToErc20.IAnyToERC20 = new anyToErc20ProxyPaymentNetwork();
```